Bio/SeqUtils/IsoelectricPoint.py

```diff
@@ -49,6 +142,11 @@ class IsoelectricPoint:
                  occurrences as integers, e.g. ``{"A": 3, "C": 0, ...}``.
                  Default: ``None``. If ``None``, the dic will be calculated
                  from the given sequence.
+    :pKa_scale: Different pKa scales. Need to be one of: Bjellqvist, DTASelect,
+            Dawson, EMBOSS, Grimsley, IPC2_peptide, IPC2_protein,
+            IPC_peptide, IPC_protein, Lehninger, Nozaki, Patrickios, ProMoST,
+            Rodwell, Sillero, Solomon, Thurlkill, Toseland, Wikipedia.
+            Default scale is 'IPC2_protein' (we assume the protein as an input)
```

Comment:
Although I can understand that you want your own method to be the default, I guess this is problematic for backward compatibility.

Reply:
It is not a matter of using my method as the default (if you can find/build a better model, I do not mind having it as the default, it is a normal thing in bioinformatics that after a few years some methods became obsolete). Currently, IPC2_protein/IPC2_peptide happens to be the most accurate*. I worked on isoelectric point subject for many years and developed multiple models. Biopython uses for pI calculation Henderson-Hasselbach equation with bisection as the main algorithm. Moreover, you need to set some pKa values for charged amino acids (Biopython uses Bjellqvist pKa set). Bjellqvist set is just one of the possible pKa sets you can use. I guess people are using it as it was one of the first introduced and it is widely implemented e.g. emboss. Yet, it is not the most accurate pKa set for proteins. I benchmarked this very extensively over the years, and even build new pKa sets that work better for proteins and peptides (for instance see http://ipc2.mimuw.edu.pl/results.html). For more details see: http://dx.doi.org/10.1186/s13062-016-0159-9 and http://dx.doi.org/10.1093/nar/gkab295

With the proposed commit, I want to give the users of biopython the opportunity to:
1) choose from 18 different pKa sets (not just Bjellqvist)
2) use as the default the most accurate* (currently it is IPC2_protein for proteins, but obviously as soon as a better isoelectric point prediction model will be available we should switch the default) 
 
* As mentioned above biopython uses a very simple approach (Henderson-Hasselbach equation with bisection), but more complicated and more accurate (and at the same time slower) algorithms exist. In the first table at http://ipc2.mimuw.edu.pl/results.html you can see models named IPC.protein.svr.19 or IPC.peptide.Conv2D that are even better than IPC2_protein, but they depend on support vector regression (sklearn) and deep learning (keras& TensorFlow). Thus my understanding is that biopython's developers want to stick to a simpler solution (Henderson-Hasselbach equation with bisection as it is pure python without any 3rd party involved), even if this compromises the accuracy a bit, rather than having dependencies like TensorFlow.